src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1140,23 +1144,34 @@ private void visitVarDeclaration(String name, JSType type) {
     }
 
     private void visitTemplateTypes(ObjectType type) {
+      visitTemplateTypes(type, Collections.<String>emptySet());
+    }
+
+    private void visitTemplateTypes(ObjectType type, Set<String> alreadyEmittedTemplateType) {
       if (type.hasAnyTemplateTypes() && !type.getTemplateTypeMap().isEmpty()) {
-        List<String> templateKeys = new ArrayList<>();
-        for (TemplateType templateType: type.getTemplateTypeMap().getTemplateKeys()) {
+        List<String> realTemplateType = new ArrayList<>();
+
+        for (TemplateType templateType : type.getTemplateTypeMap().getTemplateKeys()) {
           String displayName = templateType.getDisplayName();
+
+          // Some TemplateType are already defined in a upper level (on Type definitions for
```

Comment:
The comment is a bit hard to read. Can you mention that this is a difference in semantics between Closure and TS. In closure having @template {T} declaration both at class-level and method-level make no difference, while in typescript, the a second template variable is introduced that shadows the first one.